### ISSUE TITLE
SonnarScanner ERR_UNKNOWN_FILE_EXTENSION error

### ISSUE THREAD

**Issue by semoac** [SonarSource/sonar-css#366]
Hi!

When using the official docker images to run scanner-cli 4.6 and Sonarqube Developer 8.9, I'm getting the following error

```
INFO: Sensor CSS Rules [cssfamily]
ERROR: internal/process/esm_loader.js:74
ERROR:     internalBinding('errors').triggerUncaughtException(
ERROR:                               ^
ERROR: 
ERROR: TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension "" for /tmp/project/.scannerwork/css-bundle/bin/server
ERROR:     at Loader.defaultGetFormat [as _getFormat] (internal/modules/esm/get_format.js:71:15)
ERROR:     at Loader.getFormat (internal/modules/esm/loader.js:102:42)
ERROR:     at Loader.getModuleJob (internal/modules/esm/loader.js:231:31)
ERROR:     at async Loader.import (internal/modules/esm/loader.js:165:17)
ERROR:     at async Object.loadESM (internal/process/esm_loader.js:68:5) {
ERROR:   code: 'ERR_UNKNOWN_FILE_EXTENSION'
ERROR: }
ERROR: CSS rules were not executed. Failed to start server (60s timeout)
```

Can you help me understand the origin of the problem?

Thanks

**Comment by vilchik-elena** [comment]
could you provide full debug logs? I wonder which Node version is used

**Comment by passerbyid** [comment]
Related discussion: https://community.sonarsource.com/t/sonar-scanner-cli-failing-to-analyze-css/35562

**Comment by vilchik-elena** [comment]
It seems like a severe problem with many users suffering from it. Still it's not clear what makes Node load `bin/server` script as a module. 
I didn't manage to reproduce this problem on any version of node (testing on Mac), so may be depends on the system or environment.
Possible solution would be to add `.js` extension (https://github.com/nodejs/node/blob/master/lib/internal/modules/esm/get_format.js#L20) which is a valid ES module extension, but that would be a fix without clear understanding of the problem.

**Comment by semoac** [comment]
Hi! I'm using `docker pull sonarsource/sonar-scanner-cli:4.6`.

```
➜ docker run -ti sonarsource/sonar-scanner-cli:4.6 node -v
v14.17.1
```

Using adding `.js` to `bin/server` solve execution problem, but is hard to hack on a pipeline.

**Comment by saberduck** [comment]
the issue happens when the analyzed project is using `type: module` in `package.json`. We fixed this problem in https://github.com/SonarSource/SonarJS/pull/2878  since now CSS analysis will be provided by `SonarJS` plugin in the upcoming release of SonarQube and SonarCloud.

I am closing this issue, as we will not develop this plugin anymore.

**Comment by passerbyid** [comment]
Hi @saberduck, thanks for the fix!

Would it come in the v8.9 LTS?

**Comment by saberduck** [comment]
@passerbyid I triggered a discussion if we are going to do a bugfix for LTS version. Would it be possible to use a workaround and remove `type` field from `package.json` for you?

**Comment by passerbyid** [comment]
@saberduck 

After removing the `"type"="module"` and change all `.js` to `.mjs`, it works now.

**Comment by saberduck** [comment]
after further thinking, we have decided to not backport support for module projects. I would recommend upgrading to the latest SonarQube release to benefit from the latest features and bug fixes. This issue is fixed in SQ 9.2
